test/unit_test/components/arm_test.dart

```diff
@@ -0,0 +1,290 @@
+      server = Server([service]);
+      await server.serve(port: port);
+    });
+
+    tearDown(() async {
+      await channel.shutdown();
+      await server.shutdown();
+    });
+
+    group('Arm Service Tests', () {
+      test('endPosition', () async {
+        final client = ArmServiceClient(channel);
+        final request = GetEndPositionRequest(name: name);
+        final response = await client.getEndPosition(request);
+        expect(response.pose, arm.armEndPosition);
+      });
+
+      test('jointPositions', () async {
+        final client = ArmServiceClient(channel);
+        final request = GetJointPositionsRequest(name: name);
+        final response = await client.getJointPositions(request);
+        expect(response.positions, arm.armJoinPositions);
+      });
+
+      test('moveToJointPositions', () async {
+        final client = ArmServiceClient(channel);
+        final expected = JointPositions(values: [1, 1, 1]);
+        final request = MoveToJointPositionsRequest(name: name, positions: expected);
+        await client.moveToJointPositions(request);
+        expect(arm.armJoinPositions, expected);
+      });
+
+      test('moveToPosition', () async {
+        final client = ArmServiceClient(channel);
+        final expected = Pose(x: 1, y: 1, z: 1);
+        final request = MoveToPositionRequest(name: name, to: expected);
+        await client.moveToPosition(request);
+        expect(arm.armEndPosition, expected);
+      });
+
+      test('stop', () async {
+        final client = ArmServiceClient(channel);
+        final request = StopRequest(name: name);
+        expect(await arm.isMoving(), false);
+
+        arm.isStopped = false;
+        expect(await arm.isMoving(), true);
+
+        await client.stop(request);
+        expect(await arm.isMoving(), false);
+      });
+
+      test('isMoving', () async {
+        final client = ArmServiceClient(channel);
+        final request = IsMovingRequest(name: name);
+
+        var response = await client.isMoving(request);
+        expect(response.isMoving, false);
+
+        arm.isStopped = false;
+        response = await client.isMoving(request);
+        expect(response.isMoving, true);
+
+        arm.isStopped = true;
+        response = await client.isMoving(request);
+        expect(response.isMoving, false);
+      });
+
+      test('doCommand', () async {
+        final cmd = {'foo': 'bar'};
+
+        final client = ArmServiceClient(channel);
+        final resp = await client.doCommand(DoCommandRequest(name: name, command: cmd.toStruct()));
+        expect(resp.result.toMap()['command'], cmd);
+      });
+
+      test('extra', () async {
+        expect(arm.extra, null);
+
+        final client = ArmServiceClient(channel);
+        await client.stop(StopRequest(name: name, extra: {'foo': 'bar'}.toStruct()));
+        expect(arm.extra, {'foo': 'bar'});
+      });
+    });
+    group('Arm Client Tests', () {
```

Comment:
nit: Is there supposed to be a line before this?

Reply:
for consistency yes you're right, thank you!